### DISCUSSION TITLE
Support update scripts

### DISCUSSION THREAD

**Discussion by blitz** [renovatebot/renovate#11236] [Suggest an Idea]
### What would you like Renovate to be able to do?

We use a couple of dependency managers that are currently not supported by Renovate ([niv](https://github.com/nmattia/niv), custom internal ones, ...). While eventually it would be nice to have first-party support for this, a stop-gap measure to support pretty much anything is to allow users to provide a `update.sh` script and if it changed anything in the repository to open a PR.

For `niv` in particular, this script would just execute `niv update` and if this results in any changes to the repo, this needs to be opened as a PR.

With a lot of pain we could probably get the `regex` module to do what we want, but this is fragile, because we need to hardcode its config file format.

We have different usecases where dependencies are updated manually via script and these would also benefit from such a feature.

### If you have any ideas on how this should be implemented, please tell us here.

I think a new "update-script" manager with a configurable regex for the script location would be nice as a user.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
I would suggest you write a GitHub Action which runs a command, does a `git status`, and then creates a PR (there are actions existing to automate the last bit).

What you propose is a bit too generic for Renovate. It wouldn't understand what the dependency is, what the versions are, just "there's changed files in this repository so I'd better make a PR". It doesn't really fit in with the Renovate model

**Reply by blitz** [reply]
We have projects on Gitlab and Github. :) Each project would have to rebuild the functionality per platform. Renovate also already has the functionality to rate limit PRs and all the other nice things.

I do understand that this functionality would not be as polished as the existing features. I'll see whether I can find someone with JS knowledge (we are a C++/Rust shop) to give native support for at least `niv` a shot. This should be reasonably simple.
